routes/routes_impl.py

```diff
@@ -615,6 +622,45 @@ def bathymetry_impl(projection: str, zoom: int, x: int, y: int):
         img = plotting.tile.bathymetry(projection, x, y, zoom, {})
         return _cache_and_send_img(img, f)
 
+def mbt_impl(projection: str, tiletype: str, zoom: int, x: int, y: int):
+  """
+       Serves mbt files
+  """
+  cache_dir = current_app.config['CACHE_DIR']
+  shape_file_dir = current_app.config['SHAPE_FILE_DIR']
+  directory = str(os.path.join(cache_dir, request.path[1:]))
+  # This is a funky way of splitting the string to only include characters up to the 10th instance of '/'
+  basedir = "/".join(directory.split("/", 10)[:10])
+
+  # Send blank tile if conditions aren't met
+  if (zoom < 7) or (projection != "EPSG:3857"):
+    return send_file(shape_file_dir + "/blank.mbt")
+
+# Send file if cached or select data in SQLite file
+  if os.path.isfile(directory):
+    return send_file(directory)
+  else:
+    y = (2**zoom-1) - y
+    connection = sqlite3.connect(shape_file_dir + "/{}.mbtiles".format(tiletype))
+    selector = connection.cursor()
+    sqlite = "SELECT tile_data FROM tiles WHERE zoom_level = {} AND tile_column = {} AND tile_row = {}".format(zoom, x, y)
+    selector.execute(sqlite)
+    tile = selector.fetchone()
+    if tile == None:
+        return send_file(shape_file_dir + "/blank.mbt")
+
+    # Write tile to cache and send file
+    if os.path.isdir(basedir):
+      pass
+    else:
+      os.makedirs(basedir)
+    with open(directory + ".pbf", 'wb') as f:
+      f.write(tile[0])
+    with gzip.open(directory + ".pbf", 'rb') as gzipped:
+      with open(directory, 'wb') as tileout:
+        shutil.copyfileobj(gzipped, tileout)
```

Comment:
If I'm reading this code correctly, we're:
* Writing out the fetched tile from sqlite as a `.pbf`
* Reading it back in through gzip.
* Copying the contents of that `.pbf` into the requested tile file.

Do I have this correct?

Reply:
Yes, that's what's happening. If you have a nicer way off the top of your head I'm all ears

Reply:
I'm wondering can we just write directly to the tile file and skip the pbf entirely?

Reply:
Seems like gzip doesn't like bytesio objects for some reason. I'll create an issue and we can look at this later.